validation/src/main/java/org/ehrbase/validation/terminology/ItemValidator.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2019 Vitasystems GmbH and Christian Chevalley (Hannover Medical School).
+ *
+ * This file is part of project EHRbase
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehrbase.validation.terminology;
+
+import com.nedap.archie.rm.RMObject;
+import com.nedap.archie.rm.datavalues.quantity.DvOrdered;
+import org.ehrbase.terminology.openehr.TerminologyInterface;
+import org.ehrbase.terminology.openehr.implementation.AttributeCodesetMapping;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ItemValidator {
+
+    private Map<String, ValidationHandler> validationRegistryList;
+
+    private static final String VALIDATOR_PACKAGE = "org.ehrbase.validation.terminology.validator";
+
+    public ItemValidator() {
+        validationRegistryList = new HashMap<>();
+   }
+
+   public ItemValidator add(Class rmObjectClazz) throws NoSuchMethodException, IllegalAccessException, ClassNotFoundException {
+       Class validationHandlerClass = Class.forName(VALIDATOR_PACKAGE+"."+rmObjectClazz.getSimpleName());
```

Comment:
Suggestion:
Instead of using names to map  RMclasses to it corresponding TerminologyCheck it would be better if each TerminologyCheck  has a getRMClass which returns the RMclass it is for and use that as matching.

Reply:
fixed with the new commit